analyzer/evmverifier/evmverifier.go

```diff
@@ -0,0 +1,183 @@
+package evmverifier
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	ethCommon "github.com/ethereum/go-ethereum/common"
+
+	"github.com/oasisprotocol/oasis-indexer/analyzer"
+	"github.com/oasisprotocol/oasis-indexer/analyzer/evmverifier/sourcify"
+	"github.com/oasisprotocol/oasis-indexer/analyzer/queries"
+	"github.com/oasisprotocol/oasis-indexer/common"
+	"github.com/oasisprotocol/oasis-indexer/config"
+	"github.com/oasisprotocol/oasis-indexer/log"
+	"github.com/oasisprotocol/oasis-indexer/storage"
+	"github.com/oasisprotocol/oasis-indexer/storage/client"
+)
+
+const evmContractsVerifierName = "evmcontracts-verifier"
+
+var evmRuntimes = []common.Runtime{common.RuntimeEmerald, common.RuntimeSapphire}
+
+// evmContractsVerifier is an analyzer that verifies EVM contracts.
+type evmContractsVerifier struct {
+	interval time.Duration
+	target   storage.TargetStorage
+	logger   *log.Logger
+	chain    common.ChainName
+	client   *sourcify.SourcifyClient
+}
+
+var _ analyzer.Analyzer = (*evmContractsVerifier)(nil)
+
+func (a *evmContractsVerifier) Name() string {
+	return evmContractsVerifierName
+}
+
+type unverifiedContract struct {
+	Addr                  string
+	AddrContextIdentifier string
+	AddrContextVersion    int
+	AddrData              []byte
+	EthAddress            ethCommon.Address
+}
+
+func (a *evmContractsVerifier) getUnverifiedContracts(ctx context.Context, runtime common.Runtime) ([]*unverifiedContract, error) {
+	var unverifiedContracts []*unverifiedContract
+	rows, err := a.target.Query(ctx, queries.RuntimeEVMUnverfiedContracts, runtime)
+	if err != nil {
+		return nil, fmt.Errorf("querying unverified contracts: %w", err)
+	}
+	defer rows.Close()
+	for rows.Next() {
```

Comment:
I'm a little worried about how this will scale in the future. Maybe we can add a warning just in case? Like, if we hit the 100_000th row, log something like `"Unexpectedly high number of unverified contracts. Consider refactoring the analyzer."`.
It's not likely we'll ever hit it, but if we do, it will be a good warning to have before DB starts timing out.

Reply:
Good idea, will add the warning